File: 6TH_SEMESTER/SWE 4604 - SOFTWARE TESTING AND QUALITY ASSURANCE LAB/6/testinglab06/src/tests/FavoritesPage.java
package tests;

import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

//210042112

public class FavoritesPage extends BaseTest{


    @Test
    public void TestFavoritesAfterLogin()
    {
        //1st : login
        driver.get("https://practicesoftwaretesting.com/auth/login");

        WebElement email = driver.findElement(By.id("email"));
        email.sendKeys("customer@practicesoftwaretesting.com");

        WebElement password = driver.findElement(By.id("password"));
        password.sendKeys("welcome01");

        WebElement loginbtn = driver.findElement(By.className("btnSubmit"));
        loginbtn.click();

        //2nd: add product to favorite
        WebElement home =  driver.findElement(By.cssSelector("a[data-test='nav-home']"));
        home.click();

        WebElement clawHammer = driver.findElement(By.xpath("//h5[@data-test='product-name' and normalize-space(text())='Claw Hammer']"));
        clawHammer.click();

        WebElement addtofavoritebtn = driver.findElement(By.id("btn-add-to-favorites"));
        addtofavoritebtn.click();

    }
}
// -_- N4M154 -_-

File: 6TH_SEMESTER/SWE 4604 - SOFTWARE TESTING AND QUALITY ASSURANCE LAB/6/Lab06CSharp/TestProject1/FavoritesPage.cs
using OpenQA.Selenium;

namespace Lab06
{
    public class FavoritesPage : BaseTest
    {
        [Fact]
        public void TestFavoritesAfterLogin()
        {
            //1st : login
            driver.Navigate().GoToUrl("https://practicesoftwaretesting.com/auth/login");

            var email = driver.FindElement(By.Id("email"));
            email.SendKeys("customer@practicesoftwaretesting.com");

            var password = driver.FindElement(By.Id("password"));
            password.SendKeys("welcome01");

            var loginbtn = driver.FindElement(By.ClassName("btnSubmit"));
            loginbtn.Click();

            //2nd: add product to favorite
            var home = driver.FindElement(By.CssSelector("a[data-test='nav-home']"));
            home.Click();

            var clawHammer = driver.FindElement(By.XPath("//h5[@data-test='product-name' and normalize-space(text())='Claw Hammer']"));
            clawHammer.Click();

            var addtofavoritebtn = driver.FindElement(By.Id("btn-add-to-favorites"));
            addtofavoritebtn.Click();

        }
    }
}
